### PR TITLE
Make it easier to provide LSP configurations.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

**Comment by floitsch** [comment]
The idea is, that open-source users should add the following configuration to their settings:
```
    "toitLanguageServer.command": ["<path-to-toitlsp>", "--toitc", "<path-to-toitc>"],
```

File: vscode/src/extension.ts [hunk 1]
```diff
@@ -16,48 +16,138 @@ import { createSerialProvision } from "./toitProvision";
...
+      cliExec = configCli;
+      cliVersion = check.output;
+    }
+  }
+  if (configLspCommand !== null) {
+    lspCommand = configLspCommand as Array<string>;
+  }
+
+  if (cliExec === null && lspCommand === null) {
+    await missingCLIPrompt(configCli as string);
```
Comment (lask) [inline vscode/src/extension.ts:129]: How are we sure that `configCli` is a string at this point, and why is it necessary to have the `as`? Afterall `missingCLIPrompt` accepts null or string.

File: vscode/src/extension.ts [hunk 2]
```diff
@@ -16,48 +16,138 @@ import { createSerialProvision } from "./toitProvision";
...
+  } else if (cliExec !== null) {
+    if (cliVersion === null) {
+      await badCLIPrompt(configCli as string);
```
Comment (lask) [inline vscode/src/extension.ts:132]: Same question as above.

**Review by lask — APPROVED** [review]
lgtm

File: vscode/src/extension.ts [hunk 3]
```diff
@@ -16,48 +16,138 @@ import { createSerialProvision } from "./toitProvision";
...
+  const configCli = Workspace.getConfiguration("toit").get("path");
+  const configLspCommand = Workspace.getConfiguration("toitLanguageServer").get("command");
+
+  let cliExec: string|null = null;
+  let lspCommand: Array<string>|null = null;
+  let cliVersion: string|null = null;
+
+  if (configCli === null && configLspCommand === null) {
+    // No configuration. We try to find it in the PATH.
+    const check = await run("toit", [ "version", "-o", "short" ]);
```
[the remark this answers is not in the document]
Comment (floitsch) [reply vscode/src/extension.ts]: Created constants there that we use.
This way we only need to change one file.

File: vscode/src/extension.ts [hunk 4]
```diff
@@ -16,48 +16,138 @@ import { createSerialProvision } from "./toitProvision";
...
+    if (check.executableExists) {
+      cliExec = "toit";
+      cliVersion = check.output;
+    } else {
+      // Try to find the language server in the path.
+      const lspResult = await run("toitlsp", ["version"]);
+      if (lspResult.executableExists && lspResult.output !== null) {
+        const toitcResult = await run("toitc", ["--version"]);
+        if (toitcResult.executableExists && toitcResult.output !== null) {
+          // Good enough for us.
```
[the remark this answers is not in the document]
Comment (floitsch) [reply vscode/src/extension.ts]: done.

File: vscode/src/extension.ts (hunk 1 above)
Comment (floitsch) [reply vscode/src/extension.ts:156]: Added checks and error message.

File: vscode/src/extension.ts (hunk 2 above)
Comment (floitsch) [reply vscode/src/extension.ts:159]: added check and error message.

**Pull request merged**